src/Initializer/InitProcedure/InitMesh.cpp

```diff
@@ -107,25 +107,44 @@ static void readMeshPUML(const seissol::initializer::parameters::SeisSolParamete
     _eh(H5Pset_fapl_mpio(plist_id, seissol::MPI::mpi.comm(), info));
     hid_t dataFile =
         _eh(H5Fopen(seissolParams.mesh.meshFileName.c_str(), H5F_ACC_RDONLY, plist_id));
-    hid_t boundaryDataset = _eh(H5Dopen2(dataFile, "boundary", H5P_DEFAULT));
-    hid_t boundaryType = _eh(H5Dget_type(boundaryDataset));
-    hid_t boundarySpace = _eh(H5Dget_space(boundaryDataset));
-
-    auto boundaryTypeSize = _eh(H5Tget_size(boundaryType));
-    auto boundaryTypeRank = _eh(H5Sget_simple_extent_ndims(boundarySpace));
+    hid_t existenceTest = _eh(H5Lexists(dataFile, "boundary-format", H5P_DEFAULT));
+    if (existenceTest > 0) {
+      logInfo(rank) << "Boundary format given in PUML file.";
+      hid_t boundaryAttribute = _eh(H5Aopen(dataFile, "boundary-format", H5P_DEFAULT));
```

Comment:
wouldn't be more readable to store boundary-format as a string with I32, I64 and I32x4? 
Else
Wouldn't it make more sense to have format 0= the smallest format and 2 the largest?

Reply:
Ok, we use strings now. (however, pumgen is not up-to-date for the time being; but the attribute can also be added using h5py pretty quickly)